src/test/java/lambda/part3/exercise/TraversableTest.java

```diff
@@ -0,0 +1,167 @@
+package lambda.part3.exercise;
+
+import data.Employee;
+import data.JobHistoryEntry;
+import data.Person;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Denis Verkhoturov, mod.satyr@gmail.com
+ */
+public class TraversableTest {
+    interface Traversable<T> {
+        void forEach(Consumer<T> consumer);
+
+        static <T> Traversable<T> from(final List<T> list) {
+            return list::forEach;
+        }
+
+        default <R> Traversable<R> map(final Function<T, R> mapper) {
+            final Traversable<T> self = this;
```

Comment:
Made a mistake and threw it into another brunch with corrections ([commit](https://github.com/java8-course/lambda/pull/126/commits/c23b33be086c7920c34164fd0aba3166ed902e1e)). Now moved it here.